CONTRIBUTING.md

```diff
@@ -3,37 +3,140 @@
+
+However, whenever custom workflows are used, the promise of smooth
+updates (point 4 in the previous section) no longer holds — it is
+explicitly acceptable for ODK developers to introduce changes that may
+break custom workflows.
+
+### Technological stack and dependencies
+
+#### Docker
+
+Currently, the ODK is provided as a Docker image. While it would be
+desirable _not_ to depend on Docker, this is not realistically feasible
+for now.
+
+Still, as much as possible, developers should refrain from assuming that
+the Docker image will always be used. For example, workflows should not
+invoke a tool by hardcoding its path within the Docker image, but
+instead assume the tool is available in the _PATH_ (so, for example,
+ROBOT should be invoked simply as `robot`, _not_ as `/tools/robot`).
+
+#### Git
+
+Currently, the ODK assumes that an ontology project will be
+version-controlled using Git.
+
+There is no plan to support other version control systems (such as
+Mercurial, Subversion, etc.), and it is fine for developers to continue
+assuming the use of Git.
+
+#### GitHub
+
+Several features in the ODK assumes that an ontology project is or will
+be hosted on GitHub.
+
+This is only fine as long as those features are not *required*. It
+must *always* be possible to host a ODK-managed ontology on any other
+Git hosting service (including self-hosting).
+
+#### Operating systems and architectures.
+
+The ODK should be usable at least on:
+
+* GNU/Linux (any distribution, x86\_64 only);
+* macOS X (any version >= 10.12, x86\_64 and arm64);
+* Windows (versions 10 and 11, 86\_64 only).
+
+Running on other systems, versions, or architectures may be possible but
+is not officially supported.
+
+## Templating system
+
+Creating (“seeding”) a ODK-managed repository is done with the
+[odk/odk.py](odk/odk.py) script (available, within the ODK image, as
+`/tools/odk.py`). The `seed` command of that script instantiates the
+Jinja2 templates found in the [template/](template) directory
+(`/tools/templates` within the ODK image).
 
 For example, the file
 [template/src/ontology/Makefile.jinja2](template/src/ontology/Makefile.jinja2)
 will compile to a file `src/ontology/Makefile` in the target/output
 directory.
 
-Jinja2 templates should be fairly easy to grok for anyone familiar
-with templating systems. The syntax is very similar to Liquid
-templates, which are used extensively on the OBO site. We feed the
-template engine with a project object that is passed in by the user
-(more on that later).
+Jinja2 templates should be fairly easy to grok for anyone familiar with
+templating systems. We feed the template engine with a project object
+that is passed in by the user (see below).
+
+Logic in the templates should be kept to a minimum (though the
+aforementioned `Makefile.jinja2` template is a great offender of this
+principle).
 
-Logic in the templates should be non-existent.
+### Some notes on templating and logic
```

Comment:
@matentzn This entire section needs to be reconsidered. It is self-contradictory: we cannot simultaneously say that the logic in the Python odk code must be kept to the absolute minimum, _and_ that the templates must be as simple as possible. The logic must be _somewhere_, if it is not in the Python code then it must be in the templates, or the other way around.

I personally think we should in fact put _more_ logic into the Python code, in order to _really_ keep the templates as simple as possible.

Reply:
This is some old text, I agree with your changes and suggestions!